### DISCUSSION TITLE
店売り通常アイテムの在庫を多めに持ってほしい

### DISCUSSION THREAD

**Discussion by daradarach** [hengband/hengband#221] [Ideas]
魔力充填の巻物が売ってなくて一週間くらい宿屋で寝て過ごすことのない流通の充実した世の中になって欲しい。

**Comment by deskull-m** [comment]
単純なストレス源は解消していく路線で行きたいと思います。承知しました。

**Reply by deskull-m** [reply]
この件「店舗商品入れ替え処理の向上 #722 #723」で改善されたと思いますがいかがでしょうか。Alpha17リリース後で問題なければ完了したいと思います。

**Reply by daradarach** [reply]
魔法の地図の巻物を買うために三日くらい寝て過ごすことはぼちぼちありますが運が悪いだけなのか…ってところです。

**Reply by deskull-m** [reply]
遅れました。まだ十分でないとみなして、下記含めてもう一度問題提起しましょう。

**Reply by daradarach** [reply]
言い出しておいてなんですが、何でもかんでも常用アイテム常時店売りだと違うかなってなってきちゃうので現状程度で行きますって判断もあると思います

**Comment by daradarach** [comment]
利用頻度の高い杖のうち、鑑定の杖と周辺感知の杖は常時販売で非常にありがたいんですが、
アイテム感知の杖と邪悪存在感知の杖もかなり使う割に在庫欠品が多く、序盤から1週間ほど寝て過ごすことがあるので、
この2種も常時販売だとめちゃくちゃ嬉しいです。

**Comment by Hourier** [comment]
何が常用アイテムであるべきかをスタート地点に据えて話した方がいい感じもします
あくまで例えばですが、私は「探索楽しいなー」とか「ザック狭いなー」とか思ってアイテム感知を持ち歩いてないです
逆に、適当に酸のボール/ブレスでアイテム破壊を試みることをしないので、鑑定手段は豊富に欲しいと思ってます

あと店主の在庫も限りがあるので、常用アイテムを豊富に揃えすぎるとそれ以外のアイテムが店に並ばなくなったり売る時に難儀するかもしれません
他に例えば、呪文書の店なら常時全属性 (秘術以外2冊、秘術4冊)置いてあってもスペースがギリギリ足りるような気もします(未計算)

**Reply by iks3** [reply]
お店のページ数は増やせるので、売るスペースを増やすのは可能です。
ただ、たくさんお店のスロットがあるとランダムでいいアイテムが出る可能性が増えることになるので、バランスが難しいですね。

**Comment by chikatei** [comment]
alpha35でプレイした際、以下のアイテムがなかなか手に入らず宿屋で日数送りをすることが何度かあったので、これらも店の常備品になってほしいです。
何を常備しておいてほしいかはプレイスタイルにもよると思いますが、一プレイヤーの意見として記してみようと思います。

寺院
・致命傷の治癒の薬
・聖唱歌の巻物
・耐火の薬
・耐冷の薬

錬金術の店
・魔法の地図の巻物（固定クエに持ち込む）
・パニックモンスターの巻物

魔法の店
・邪悪感知の杖（自前で感知できない職で中盤まで多用）
・アイテム感知の杖

また、杖は回数でスタックがバラけるのでわかりますが、致命傷の治癒の薬や魔力充填の巻物などは常に99個とかそれに類する数量のスタックで店に生成されてもいいような気がします。店に並んでいても10個しか置いてないとかだと結局必要な数揃えるためには日数送りすることがありました。

**Comment by ghost** [comment]
久々にプレー(alpha48)してちょっとアイテム在庫の巡りが悪いなあと思う事が多かったので。
これまでの議論で店のアイテムページを増やすのはバランスを崩しかねないということでしたので、
「店は常にフル充填の杖・魔法棒を売る」
という仕様にすることで充填回数違いの同種杖・魔法棒がお店のスロットを消費するのを防ぐというのはどうでしょうか？(魔法の店、BMぐらいしか恩恵がありませんが)
割引による同種アイテム併売は序盤の金欠時にありがたいのですが、回数違いによる価格差はさほどでもなく今一つプレー上のメリットも感じられません。
以上、ご検討いただければ幸いです。

例）フル充填(21回) のみの販売にすれば↓のようなケースで鑑定の杖は１スロット消費で済む
![image](https://user-images.githubusercontent.com/79756503/147364614-e1910e99-d805-4931-bde0-ca155c6c9f72.png)

**Reply by ghost** [reply]
追記
あくまで店頭の新規販売分についてを念頭に書いております。
@が売った魔法棒・杖については現行通り、その時点の充填回数で店頭に並ぶのが自然と思います。
